### PR TITLE
Minor: Consolidate UDF tests

### PR DESCRIPTION
**Pull request by alamb**
## Rationale for this change

When evaluating coverage of our user defined functions (UDFs) I found that some tests were in sql_integration, and some were in other places.

## What changes are included in this PR?

Consolidate them all into the aptly named `user_defined_integration` so they are easier to find.  

## Are these changes tested?
all tests

### CODE REVIEW

File: datafusion/core/tests/user_defined/user_defined_scalar_functions.rs [hunk 1]
```diff
@@ -15,26 +15,40 @@
...
 #[tokio::test]
 async fn csv_query_custom_udf_with_cast() -> Result<()> {
-    let ctx = create_ctx();
+    let ctx = create_udf_context();
     register_aggregate_csv(&ctx).await?;
     let sql = "SELECT avg(custom_sqrt(c11)) FROM aggregate_test_100";
-    let actual = execute(&ctx, sql).await;
-    let expected = vec![vec!["0.6584408483418833"]];
-    assert_float_eq(&expected, &actual);
+    let actual = plan_and_collect(&ctx, sql).await.unwrap();
```
Comment (alamb) [inline datafusion/core/tests/user_defined/user_defined_scalar_functions.rs:41]: this was using a (very) old style to execute

File: datafusion/core/src/execution/context.rs [hunk 2]
```diff
@@ -2330,87 +2330,6 @@ mod tests {
         Ok(())
     }
 
-    #[tokio::test]
```
Comment (alamb) [inline datafusion/core/src/execution/context.rs:2333]: moved to user defined tests

File: datafusion/core/tests/sql/expr.rs [hunk 3]
```diff
@@ -886,18 +886,6 @@ async fn csv_query_nullif_divide_by_0() -> Result<()> {
     Ok(())
 }
 
-#[tokio::test]
```
Comment (alamb) [inline datafusion/core/tests/sql/expr.rs:889]: this was randomly in the file for testing exprs when it is actually an (user defined) aggregate query  😕

File: datafusion/core/tests/sql/mod.rs [hunk 4]
```diff
@@ -101,54 +96,6 @@ pub mod select;
 mod sql_api;
 pub mod subqueries;
 pub mod timestamp;
-pub mod udf;
-
-fn assert_float_eq<T>(expected: &[Vec<T>], received: &[Vec<String>])
```
Comment (alamb) [inline datafusion/core/tests/sql/mod.rs:106]: not needed

**Review by yjshen — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @yjshen !

**Pull request merged**